### PR TITLE
Update services structure

### PR DESCRIPTION
**Pull request by alealvarez14**
## Pull request type

- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

[Update services structures](https://github.com/rootstrap/node-ts-api-base/projects/1#card-52762759)

Issue Link: N/A

## What is the new behavior?

-

## Other information

N/A

## Preview

N/A

### CODE REVIEW

File: src/controllers/auth.controller.ts [hunk 1]
```diff
@@ -10,10 +10,16 @@ import {
...
 import * as _ from 'lodash';
 import { User } from '@entities/user.entity';
-import { createJWT } from '@services/jwt.service';
+import JWTService from '@services/jwt.service';
 
 @JsonController('/auth')
 export class AuthController {
+  jwt: JWTService;
+
+  constructor(private jwtService: JWTService) {
```
Comment (kevquincke) [inline src/controllers/auth.controller.ts]: I think only adding it to the constructor will work, and you can access it as `this.jwtService` can you verify that?

File: src/services/jwt.service.ts [hunk 2]
```diff
@@ -1,44 +1,46 @@
...
-        { expiresIn: ACCESS_TOKEN_LIFE || '6h' }
-      );
-      resolve(token);
-    } catch (error) {
-      reject(new Error('Error creating JWT'));
-    }
-  });
-};
+@Service()
+export default class {
```
Comment (kevquincke) [inline src/services/jwt.service.ts]: Can we name this class JWTService or something like that?

File: __tests__/controllers/users.controller.spec.ts [hunk 3]
```diff
@@ -4,7 +4,9 @@ import { factory, useSeeding } from 'typeorm-seeding';
 import app from '@app';
 import connection from '@database/connection';
 import { User } from '@entities/user.entity';
-import { createJWT } from '@services/jwt.service';
+import 'reflect-metadata';
```
Comment (kevquincke) [inline __tests__/controllers/users.controller.spec.ts]: is this line necessary? since it's already imported in the `app`

**Review by kevquincke — APPROVED** [review]
Good one! Left some comments

File: __tests__/controllers/users.controller.spec.ts (hunk 3 above)
Comment (alealvarez14) [reply __tests__/controllers/users.controller.spec.ts]: Nope, you're right!

File: src/services/jwt.service.ts (hunk 2 above)
Comment (alealvarez14) [reply src/services/jwt.service.ts]: Yes, I'll do the same for the S3 service class as well.

**Review by Alxzu — APPROVED** [review]

**Pull request merged**